frontend/src/features/search/Search.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+
+import { loadSearchResults } from './state/searchActions'
+import { selectSearchResults } from './state/searchState'
+import NavBar from '../navbar/NavBar';
+import BigSearchBox from './BigSearchBox';
+import SearchResultItem from './SearchResultItem'
+import PageControl, { PageChangeObject } from '../../chrome/PageControl'
+import Footer from '../footer/Footer'
+
+import { QueryParams } from '../../qri/queryParams'
+
+interface SearchProps {
+  queryParams: QueryParams
+}
+
+const Search: React.FC<SearchProps> = ({ queryParams }) => {
+  const dispatch = useDispatch()
+  const history = useHistory()
+  const searchResults = useSelector(selectSearchResults)
+
+  // keep track of the current query and page in local state
+  // TODO(chriswhong): add sorting
+  const [q, setQ] = useState(queryParams.q)
+  const [page, setPage] = useState(queryParams.page)
+
+  const prevQ = useRef()
+
+  // TODO(chriswhong): get pageInfo from state, this is a placeholder to
+  // lay out and test PageControl
+  const pageInfo = {
+    resultCount: 68,
+    page,
+    pageSize: 25,
+    pageCount: 3
+  }
+
+  // on first load, or when local state changes,
+  // update query params and fetch new data
+  useEffect(() => {
+  const params = new URLSearchParams()
+    if (q) {
+      params.append('q', q)
+    } else {
+      params.delete('q')
+    }
+
+    if (page) {
+      params.append('page', page.toString())
+    } else {
+      params.delete('page')
+    }
+
+    if (q) {
+      console.log('effect', q, prevQ.current)
+      // if q has changed, reset page to 1
+      if (prevQ.current && (q !== prevQ.current)) {
+        setPage(1)
+        prevQ.current = q
+      } else {
+        // update the current route with new search params
+        history.push({ search: params.toString() })
+
+        // go get new data
+        dispatch(loadSearchResults(q, page))
+      }
+    }
+
+  }, [q, page, history, dispatch])
```

Comment:
this needs to be rethought & integrated with the state tree. Let's pair up & figure out the right solution once other issues in this PR are addressed.

Reply:
Refactored!